### ISSUE TITLE
CPPD decoder - mistake copy and paste

### ISSUE THREAD

**Issue by FahNos** [Topdu/OpenOCR#112]
Dear master,
I check the code of CPPD decoder and see have mistake from line 224 to 233:
`self.edge_decoder = nn.ModuleList([
            DecoderLayer(
                dim=dim,
                num_heads=dim // 32 if num_heads is None else num_heads,
                mlp_ratio=4.0,
                qkv_bias=True,
                qk_scale=1.0 if (rec_layer + i) % 2 != 0 else None,
                drop_path=dpr[num_layer + i],
            ) for i in range(rec_layer)
        ])`
I think you should use class EdgeDecoderLayer(nn.Module) instead of class DecoderLayer(nn.Module)

I check the code again and see the different between forward_train and forward_test in Class CPPDhead,
- In forward train:
`char_vis_node_query = paddle.concat([char_node_embed, visual_feats], 1)
char_vis_node_query = char_decoder_layer(
                char_vis_node_query, char_vis_node_query[:, counting_char_num:, :]
            )
char_vis_feats = char_vis_node_query[:, counting_char_num:, :]
pos_node_feats = self.edge_decoder(
            pos_node_query, char_vis_feats, char_vis_feats
        )`
- In forward_test:
`char_vis_node_query = visual_feats
char_vis_node_query = char_decoder_layer(
                char_vis_node_query, char_vis_node_query
            )
char_vis_feats = char_vis_node_query

        pos_node_feats = self.edge_decoder(
            pos_node_query, char_vis_feats, char_vis_feats
        )`
We can see in forward train, tensor char_vis_node_query is attention between concat([char_node_embed, visual_feats], 1) and visual_feats, but in forward test char_vis_node_query is attention between visual_feats and visual_feats, this can make the different between char_vis_feats = char_vis_node_query[:, counting_char_num:, :] in forward_train and char_vis_feats = char_vis_node_query in forward_test

One more think in forward train I see have use
`diag_mask = (
            paddle.eye(pos_node_feats1.shape[1])
            .unsqueeze(0)
            .tile([pos_node_feats1.shape[0], 1, 1])
        )`
but in forward_test do not use it

Do you test it and see the different between train and test process ?
If result is good, Can you explain why have different but the result still good
